examples/e2e/config.py

```diff
@@ -0,0 +1,81 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
```

Comment:
This is very interesting, I think some time ago we were already discussing whether/how to make configuration management part of ZenML itself. Might also solve/help with the hyperparam tuning issue. Perhaps we could revisit this now that you're on the team!

Reply:
I put some thinking into the configuration topic and didn't find a better way to do it rather than using class as now.
It can be relatively simple to handle step-level configs with `extra`, but things are getting messed up once we need access to config during pipeline composition.
For the use-case of HP tuning, I generate a set of steps to run the search in parallel based on the config and also rely on it inside the step, because serializing sklearn, scipy, numpy, etc objects of all kinds is super tricky. The search grid for HP tuning might even have some lambdas or whatever else.

So, I prefer to keep this as-is here for now and try to shape better how we deal with configurations.
Thoughts - @fa9r @bcdurak @htahir1 ?

Reply:
@avishniakov Can you make a framework PRD in the framework board and add basic details and context to it? Then put it in a rough timeline for next quarter . We’ll take it up when the time is right